lib/metasploit/framework/login_scanner/ssh.rb

```diff
@@ -162,8 +162,10 @@ def gather_proof
                   proof = ssh_socket.exec!("systeminfo\n").to_s
                   /OS Name:\s+(?<os_name>.+)$/ =~ proof
                   /OS Version:\s+(?<os_num>.+)$/ =~ proof
-                  if os_name && os_num
-                    proof = "#{os_name.chomp} #{os_num.chomp}"
+                  if os_num.present? && os_name.present?
+                    proof = "#{os_name.strip} #{os_num.strip}"
+                  else
+                    proof = ssh_socket.exec!("ver\n").strip
```

Comment:
Just realizing, I think the use of `to_s` was required in case nil is returned

```suggestion
                    proof = ssh_socket.exec!("ver\n").to_s.strip
```